src/main/java/core/db/EconomyTable.java

```diff
@@ -68,38 +70,58 @@ protected String[] getCreateIndexStatement() {
 	 */
 	void storeEconomyInfoIntoDB(int hrfId, Economy economy, Timestamp date) {
-		final String[] whereColumns = {columns[0].getColumnName()};
-		final String[] whereValues = {"" + hrfId};
-
 		if (economy != null) {
 			//first delete existing entry
-			delete(whereColumns, whereValues);
-
-			//insert new data
-			var statement =
-					"INSERT INTO "
-							+ getTableName()
-							+ " ( HRF_ID, FetchedDate, SupportersPopularity, SponsorsPopularity, Cash, IncomeSponsors, IncomeSponsorsBonus, IncomeSpectators, IncomeFinancial, IncomeTemporary, IncomeSum, CostsPlayers, " +
-							"CostsStaff, CostsArena, CostsYouth, CostsFinancial, CostsTemporary, CostsSum, ExpectedWeeksTotal, LastIncomeSponsors, LastIncomeSponsorsBonus, LastIncomeSpectators, LastIncomeFinancial, " +
-							"LastIncomeTemporary, LastIncomeSum, LastCostsPlayers, LastCostsStaff, LastCostsArena, LastCostsYouth, LastCostsFinancial, LastCostsTemporary, LastCostsSum, LastWeeksTotal, ExpectedCash, " +
-							"IncomeSoldPlayers, IncomeSoldPlayersCommission, CostsBoughtPlayers, CostsArenaBuilding, LastIncomeSoldPlayers, LastIncomeSoldPlayersCommission, LastCostsBoughtPlayers, LastCostsArenaBuilding) VALUES("
-							+ hrfId + ", '" + date.toString() + "', " + economy.getSupportersPopularity() + ", " + economy.getSponsorsPopularity() + ", "
-							+ economy.getCash() + ", " + economy.getIncomeSponsors() + ", " + economy.getIncomeSponsorsBonus() + ", "+ economy.getIncomeSpectators() + ", " + economy.getIncomeFinancial() + ", "
-							+ economy.getIncomeTemporary() + ", " + economy.getIncomeSum() + ", " + economy.getCostsPlayers() + ", " + economy.getCostsStaff() + ", "
-							+ economy.getCostsArena() + ", " + economy.getCostsYouth() + ", " + economy.getCostsFinancial() + ", " + economy.getCostsTemporary() + ","
-							+ economy.getCostsSum() + ", " + economy.getExpectedWeeksTotal() + ", " + economy.getLastIncomeSponsors() + ", " + economy.getLastIncomeSponsorsBonus() + "," + economy.getLastIncomeSpectators() + ", "
-							+ economy.getLastIncomeFinancial() + ", " + economy.getLastIncomeTemporary() + ", " + economy.getLastIncomeSum() + ", " + economy.getLastCostsPlayers() + ","
-							+ economy.getLastCostsStaff() + ", " + economy.getLastCostsArena() + ", " + economy.getLastCostsYouth() + ", " + economy.getLastCostsFinancial() + ", "
-							+ economy.getLastCostsTemporary() + ", " + economy.getLastCostsSum() + ", " + economy.getLastWeeksTotal() + ", " + economy.getExpectedCash() + ", "
-							+ economy.getIncomeSoldPlayers() + ", " + economy.getIncomeSoldPlayersCommission() + ", " + economy.getCostsBoughtPlayers() + ", " + economy.getCostsArenaBuilding() + ", "
-							+ economy.getLastIncomeSoldPlayers() + ", " + economy.getLastIncomeSoldPlayersCommission() + ", " + economy.getLastCostsBoughtPlayers() + ", " + economy.getLastCostsArenaBuilding()
-							+ ")";
-			adapter.executeUpdate(statement);
+			executePreparedDelete( hrfId);
+			executePreparedInsert(
```

Comment:
Neat! :-)

Reply:
there were only a few places that used a sql update and i thought it would be a good idea to make it unique.  do you think that it is better to use the update ?

Reply:
I would be more comfortable with update, yes.  My worry is that if for some reason, HO is killed between delete and insert, the record will be lost, and this may cause further troubles.

Reply:
OK  - what would you suggest then

- checking existance of record before inserting or updating (should be as slow as deleting;-)
-  saving status in model classes when loading record from db (isAvailableInDb)

Reply:
a day of reflection, brings me to the idea to implement the 2nd plan times

Reply:
latest commit implements the update in SPIELERTABLE (delete is removed).

one thing, that is not so good: the great similarity of update and insert method.

Reply:
with latest commit the methods store, update and insert are in AbstractTable and shorter than a limerick.